Library/Homebrew/cmd/--env.rb

```diff
@@ -3,19 +3,37 @@
 
 require "extend/ENV"
 require "build_environment"
+require "utils/shell"
 
 module Homebrew
+  def explicit_shell
+    shell = ARGV.value("shell")
+    return :noshell if shell == "none"
```

Comment:
Our `ARGV.value` has a few problems, one of them being that it doesn't accept an empty value after the equals sign (it expects at least one character). I find this surprising and undesirable, thus opened #231 to address this. You could start using `--shell=` if that goes through.

Here's a proposal (both regarding behavior and option names):

- `brew --env` preserves existing behavior, i.e. uses the colon-delimited format for TTY and `bash`-compatible format otherwise. (We want to preserve this behavior as not not break existing scripts that may rely on this.)
- `brew --env --plain` always uses the colon-delimited format (currently `--shell=none`).
- `brew --env --export[=auto]` auto-detects the shell and uses a format suitable for exporting to the detected shell.
- `brew --env --export=<shell>` uses a format suitable for exporting to the explicitly selected shell.

(Or maybe just use `--format=(plain|auto|<shell>)` to cover the latter three cases?)